DynamicSmoothedGUI.py

```diff
@@ -291,11 +343,13 @@ def updateGridSmoothed(self):
         try:
             if self.desired_heading is not None and self.heading == self.desired_heading:
                 self.draw_headings()
+                self.drawC1C0()
```

Comment:
This executes at every iteration

Reply:
except when robot is turning right?

Reply:
I'm supposing your self.desired_heading is a degree of the heading, and it is initialized to be None; if the robot is not turning then the desired_heading is None and I suppose this won't be executed(I might be wrong please correct me)? For the self.drawC1C0() function I see that it is called in some of the if cases and else cases, but not always called in the updateGridSmoothed function, therefore I didn;t see a good way to extract this function outside of the many "if cases"; it would be nice if we could improve tho

Reply:
Yes Jackie, except when turning that is correct